spec.html

```diff
@@ -42712,30 +42726,41 @@ <h1>
         </emu-note>
       </emu-clause>
 
-      <emu-clause id="sec-hostenqueueatomicswaitasynctimeoutjob" type="host-defined abstract operation">
+      <emu-clause id="sec-enqueueatomicswaitasynctimeoutjob" type="abstract operation">
         <h1>
-          HostEnqueueAtomicsWaitAsyncTimeoutJob (
+          EnqueueAtomicsWaitAsyncTimeoutJob (
             _WL_: a WaiterList Record,
             _waiterRecord_: a Waiter Record,
           ): ~unused~
         </h1>
         <dl class="header">
         </dl>
-        <p>Let _timeoutJob_ be a new Job Abstract Closure with no parameters that captures _WL_ and _waiterRecord_ and performs the following steps when called:</p>
         <emu-alg>
+        1. Let _timeoutJob_ be a new Job Abstract Closure with no parameters that captures _WL_ and _waiterRecord_ and performs the following steps when called:
           1. Perform EnterCriticalSection(_WL_).
           1. If _WL_.[[Waiters]] contains _waiterRecord_, then
             1. Let _now_ be the time value (UTC) identifying the current time.
-            1. If ℝ(_now_) ≥ _waiterRecord_.[[TimeoutTime]], then
-              1. Set _waiterRecord_.[[Result]] to *"timed-out"*.
-              1. Perform RemoveWaiter(_WL_, _waiterRecord_).
-              1. Perform NotifyWaiter(_WL_, _waiterRecord_).
-            1. Else,
-              1. Perform HostEnqueueAtomicsWaitAsyncTimeoutJob(_WL_, _waiterRecord_).
+            1. Assert: ℝ(_now_) ≥ _waiterRecord_.[[TimeoutTime]].
+            1. Set _waiterRecord_.[[Result]] to *"timed-out"*.
+            1. Perform RemoveWaiter(_WL_, _waiterRecord_).
+            1. Perform NotifyWaiter(_WL_, _waiterRecord_).
           1. Perform LeaveCriticalSection(_WL_).
           1. Return ~unused~.
+        1. Let _now_ be the time value (UTC) identifying the current time.
+        1. Perform HostEnqueueTimeoutJob(_timeoutJob_, 𝔽(_waiterRecord_.[[TimeoutTime]]) - _now_).
         </emu-alg>
-        <p>An implementation of HostEnqueueAtomicsWaitAsyncTimeoutJob schedules _timeoutJob_ to be performed at some future time. It must also conform to the requirements in <emu-xref href="#sec-jobs"></emu-xref>.</p>
+      </emu-clause>
+
+      <emu-clause id="sec-hostenqueuetimeoutjob" type="host-defined abstract operation">
+        <h1>
+          HostEnqueueTimeoutJob (
+            _timeoutJob_: a Job Abstract Closure,
+            _milliseconds_: a Number,
```

Comment:
should this be a non-negative integer?

Reply:
Good suggestion, changed it to non-negative integral Number. It's an actual number, not an integer mathematical value, because that feels more correct for the host API: we're not requiring 262 and hosts communicate the reals.